File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PutFlagSettingExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PutFlagSettingExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var valuePut = new ValuePut(
            comment: "make sure this context experiences a specific variation"
        );

        try
        {
            new UserSettingsApi(config).PutFlagSetting(
                projectKey: "projectKey_string",
                environmentKey: "environmentKey_string",
                userKey: "userKey_string",
                featureFlagKey: "featureFlagKey_string",
                valuePut: valuePut
            );
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling UserSettingsApi#PutFlagSetting: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PutFlagSettingExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const valuePut: api.ValuePut = {
  comment: "make sure this context experiences a specific variation",
};

new api.UserSettingsApi(configuration).putFlagSetting(
  "projectKey_string", // projectKey
  "environmentKey_string", // environmentKey
  "userKey_string", // userKey
  "featureFlagKey_string", // featureFlagKey
  valuePut,
).catch((error: Error | AxiosError) => {
  console.log("Exception when calling UserSettingsApi#putFlagSetting:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
